packages/block-editor/src/components/block-list/block-invalid-warning.js

```diff
@@ -23,6 +23,7 @@ export function BlockInvalidWarning( {
 	block,
 } ) {
 	const hasHTMLBlock = !! getBlockType( 'core/html' );
+	const hasClassicBlock = !! getBlockType( 'core/freeform' );
```

Comment:
I think this component should really use `canInsertBlockType`.

Reply:
Just to add, I can push a PR that does this as a follow up, as this solves the immediate issue in the UI for the user.

(I accidentally started looking at the same issue in parallel).

Reply:
Yes, that would be ideal for both `core/html` and `core/freeform` checks. However, I only decided to reuse the method since this is targeted for WP 6.2 RC.

We should also refactor away from `withDispatch|Select` hooks in the future.

Reply:
PR here - https://github.com/WordPress/gutenberg/pull/49052